.github/workflows/build-documentation.yml

```diff
@@ -1,49 +1,23 @@
-# This is a basic workflow to help you get started with Actions
-
 name: Build Documentation
 
-# Controls when the workflow will run
-on:
-  workflow_run:
-    workflows: ["Build and update artifacts"]
-    branches: [main]
-    types: 
-      - completed
-  push:
-    branches: [ main ]
-    paths:
-    - docs/*
-
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
-
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+on: push
+ 
 jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
+  build-docs:
+    name: Build documentation
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: [3.9]
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v1
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: 3.9
 
       - name: Install dependencies
-        run: |
+        run: >-
           pip install -r requirements.txt
-      
-      - name: Build docs and push to gh-pages
-        run: |
-          git pull
-          git config --local user.email "action@github.com"
-          git config --local user.name "GitHub Action"
-          make gh-deploy
+
+      - name: Build docs
+        run: >-
+          ./scripts/generate_linkml_docs.py src/schema/submission_centric_schema.yaml 
```

Comment:
Here docs are build but not committed, right? Either way, to be consistent with our other actions, it would be nice if this action would only check whether the docs are up to date (would require changes to the python script, please see my comments below). If they are not, the check should fail. Performing the update has to be done manually. While this is a bit less convenient, we have only one source of truth, the developer (and not additionally the CI system). This makes merge conflicts less likely.

Reply:
TODO in another ticket